operator/pkg/reconciliation/secrets.go

```diff
@@ -227,19 +227,23 @@ func (rc *ReconciliationContext) retrieveInternodeCredentialSecretOrCreateDefaul
 func validateCassandraUserSecretContent(dc *api.CassandraDatacenter, secret *corev1.Secret) []error {
 	var errs []error
 
-	namespacedName := types.NamespacedName{
-		Name:      secret.ObjectMeta.Name,
-		Namespace: secret.ObjectMeta.Namespace,
-	}
-	errorPrefix := fmt.Sprintf("Validation failed for user secret: %s", namespacedName.String())
-
-	for _, key := range []string{"username", "password"} {
-		value, ok := secret.Data[key]
-		if !ok {
-			errs = append(errs, fmt.Errorf("%s Missing key: %s", errorPrefix, key))
-		} else if !utf8.Valid(value) {
-			errs = append(errs, fmt.Errorf("%s Key did not have valid utf8 value: %s", errorPrefix, key))
+	if secret != nil {
+		namespacedName := types.NamespacedName{
+			Name:      secret.ObjectMeta.Name,
+			Namespace: secret.ObjectMeta.Namespace,
+		}
+		errorPrefix := fmt.Sprintf("Validation failed for user secret: %s", namespacedName.String())
+
+		for _, key := range []string{"username", "password"} {
+			value, ok := secret.Data[key]
+			if !ok {
+				errs = append(errs, fmt.Errorf("%s Missing key: %s", errorPrefix, key))
+			} else if !utf8.Valid(value) {
+				errs = append(errs, fmt.Errorf("%s Key did not have valid utf8 value: %s", errorPrefix, key))
+			}
 		}
+	} else {
+		errs = append(errs, fmt.Errorf("Validation failed due to a missing secret"))
```

Comment:
@burmanm am I right that our generic error handling will pull this string onto an event on the CassDC you can see with `kubectl describe`?

Reply:
No, currently only the first validation issue is returned as error, rest are ignored:

https://github.com/datastax/cass-operator/blob/master/operator/pkg/reconciliation/handler.go#L306

Reply:
But yes, that one error is pushed to the recorder:

https://github.com/datastax/cass-operator/blob/master/operator/pkg/reconciliation/handler.go#L247